### PR TITLE
Make egress tests workload agnostic and extend for vm workloads

### PR DESCRIPTION
**Pull request by JimmyCYJ**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: tests/integration/security/sds_tls_origination/util/util.go [hunk 1]
```diff
@@ -152,31 +172,41 @@ func createSecret(credentialType string, cn, ns string, ic TLSCredential, isNotG
 // SetupEcho creates two namespaces client and server. It also brings up two echo instances server and
 // client in respective namespaces. HTTP and HTTPS port on the server echo are set up. Egress Gateway is set up in the
 // service namespace to handle egress for "external" calls.
-func SetupEcho(t test.Failer, ctx resource.Context) (echo.Instance, echo.Instance, namespace.Instance, namespace.Instance) {
-	clientNamespace := namespace.NewOrFail(t, ctx, namespace.Config{
+func SetupEcho(t test.Failer, ctx resource.Context, apps *EchoDeployments) {
```
Comment (yangminzhu) [inline tests/integration/security/sds_tls_origination/util/util.go:175]: The comment needs some update.

File: tests/integration/security/sds_tls_origination/util/util.go [hunk 2]
```diff
@@ -152,31 +172,41 @@ func createSecret(credentialType string, cn, ns string, ic TLSCredential, isNotG
...
 	cluster := ctx.Clusters().Default()
-	var internalClient, externalServer echo.Instance
-	echoboot.NewBuilder(ctx).
+	var internalClient, internalVM, externalServer echo.Instance
+	echos := echoboot.NewBuilder(ctx).
 		With(&internalClient, echo.Config{
 			Cluster:   cluster,
-			Service:   "client",
-			Namespace: clientNamespace,
+			Service:   "a",
```
Comment (yangminzhu) [inline tests/integration/security/sds_tls_origination/util/util.go:189]: why only change the client to "a" but not the server?

File: tests/integration/security/sds_tls_origination/main_test.go [hunk 3]
```diff
@@ -20,16 +20,13 @@ import (
...
 	"istio.io/istio/pkg/test/framework/components/istio"
-	"istio.io/istio/pkg/test/framework/label"
 )
 
 var inst istio.Instance
 
 func TestMain(m *testing.M) {
 	framework.
 		NewSuite(m).
-		Label(label.CustomSetup).
```
Comment (yangminzhu) [inline tests/integration/security/sds_tls_origination/main_test.go:31]: just curious, the custom label is no longer needed?

File: tests/integration/security/sds_tls_origination/util/util.go (hunk 1 above)
Comment (JimmyCYJ) [reply tests/integration/security/sds_tls_origination/util/util.go:175]: Nice catch. Done. Thanks!

File: tests/integration/security/sds_tls_origination/util/util.go (hunk 2 above)
Comment (JimmyCYJ) [reply tests/integration/security/sds_tls_origination/util/util.go:189]: I still use single server workload in the test, and extend clients to cover vm workload. 
I would prefer to set up server on a VM in a follow up PR.

File: tests/integration/security/sds_tls_origination/main_test.go (hunk 3 above)
Comment (JimmyCYJ) [reply tests/integration/security/sds_tls_origination/main_test.go:31]: Yes, I think we can remove it as the [setupConfig()](https://github.com/istio/istio/blob/release-1.7/tests/integration/security/sds_tls_origination/main_test.go#L43) has been removed.

**Review by yangminzhu — APPROVED** [review]

**Pull request merged**